crates/bevy_derive/src/bevy_main.rs

```diff
@@ -10,19 +10,10 @@ pub fn bevy_main(_attr: TokenStream, item: TokenStream) -> TokenStream {
     );
 
     TokenStream::from(quote! {
-        #[no_mangle]
         #[cfg(target_os = "android")]
-        unsafe extern "C" fn ANativeActivity_onCreate(
-            activity: *mut std::os::raw::c_void,
-            saved_state: *mut std::os::raw::c_void,
-            saved_state_size: usize,
-        ) {
-            bevy::ndk_glue::init(
-                activity as _,
-                saved_state as _,
-                saved_state_size as _,
-                main,
-            );
+        #[cfg_attr(target_os = "android", bevy::ndk_glue::main(backtrace = "on", ndk_glue = "bevy::ndk_glue"))]
+        fn android_main() {
```

Comment:
Where is this function called / why does this exist? Presumably ndk_glue is doing some magic with it?

Reply:
yup, ndk-glue will create an activity with it, and then the activity is called on app start. This was more or less what was done before directly, but I prefer to let ndk-glue handle it

Reply:
https://github.com/rust-windowing/android-ndk-rs/blob/5747e07f05183e66a20375386deca1dcb3283848/ndk-macro/src/expand.rs#L90-L104 should look similar to what we had before

Reply:
Excellent. I much prefer this approach too. Can you drop a link the relevant ndk-glue docs in a comment here?